datasets/ua/war_sanctions/crawler.py

```diff
@@ -0,0 +1,381 @@
+            dob = dob_pob[0]
+            h.apply_date(person, "birthDate", dob)
+    if positions:
+        for position in h.multi_split(positions, [" / "]):
+            person.add("position", position)
+    person.add("topics", "poi")
+
+    sanction = h.make_sanction(context, person)
+    sanction.add("reason", " ".join(data.pop("Reasons")))
+    sanction.add("sourceUrl", link)
+    sanction.add("program", program)
+
+    context.emit(person, target=True)
+    context.emit(sanction)
+    context.audit_data(data, ignore=["Sanction Jurisdictions"])
+
+
+def crawl_legal_entity(context: Context, link, program):
+    detail_page = context.fetch_html(link, cache_days=1)
+
+    # Having at least some pop()s without defaults and audit()-ing the rest
+    # implies the very generic selectors.
+    data = {}
+    for row in detail_page.findall(".//main//div[@class='row']"):
+        divs = row.findall("div")
+        if len(divs) == 2:
+            label_elem, value_elem = divs
+            if "yellow" in label_elem.get("class"):
+                label, value = extract_label_value_pair(label_elem, value_elem, data)
+                data[label] = value
+    name = data.pop("Name")
+    name_abbr = data.pop("Abbreviated name of the legal entity", None)
+    reg_num = data.pop("Registration number")
+
+    legal_entity = context.make("LegalEntity")
+    legal_entity.id = context.make_id(name, name_abbr, reg_num)
+    legal_entity.add("name", name)
+    legal_entity.add("name", name_abbr)
+    legal_entity.add("registrationNumber", reg_num)
+    legal_entity.add("address", data.pop("Address"))
+    legal_entity.add("country", data.pop("Country"))
+    legal_entity.add("taxNumber", data.pop("Tax Number"))
+    legal_entity.add("sourceUrl", data.pop("Links"))
+    archive_links = data.pop("Archive links", None)
+    if archive_links is not None:
+        legal_entity.add("sourceUrl", archive_links)
+
+    legal_entity.add("topics", "poi")
+    sanction = h.make_sanction(context, legal_entity)
+    sanction.add("reason", " ".join(data.pop("Reasons")))
+    sanction.add("sourceUrl", link)
+    sanction.add("program", program)
+
+    context.emit(legal_entity, target=True)
+    context.emit(sanction)
+    context.audit_data(data, ignore=["Sanction Jurisdictions"])
+
+
+def extract_next_page_url(doc):
+    # next page <a> element extraction using xpath
+    next_link_element = doc.xpath("//ul[@class='pagination']//li[@class='next']/a")
+    if next_link_element:
+        next_link = next_link_element[0]
+        return next_link.get("href")
+
+    return None
+
+
+def crawl(context: Context):
+    main_page = context.fetch_html(context.data_url, cache_days=1)
+    node = main_page.find(
+        ".//section[@class='sections d-flex flex-wrap align-items-stretch justify-content-center mb-5 pl-5 pr-5 medium']"
+    )
+    api_section = main_page.find(
+        ".//div[@class='d-flex flex-column flex-sm-row menu-alt']"
```

Comment:
in the browser, these class values are not ordered. If you were writing CSS, which most or all of these are intended for, you could select this element with something like `div.sections.justify-content-center` or equally `div.justify-content-center.sections`

They can add more classes in between, rename some of these, change mb-5 to mb-4 to adjust margin ever so slightly, and now we have to come in and adjust this. Not such a big deal, but I think as a general rule, it's good to start thinking about classes in a similar way to how a web developer does. so instead of hard-matching the exact ordering of 5+ classes, let's see if we can make the selector as selective, but robust to change, as possible.